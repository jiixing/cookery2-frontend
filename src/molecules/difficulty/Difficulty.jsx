// @flow

import React from 'react'
import { Badge } from 'reactstrap'

import type { Difficulty as DifficultyType } from 'types'

const classForLevel = (level: DifficultyType) => {
  switch (level) {
    case 'easy':
      return 'success'
    case 'medium':
      return 'warning'
    default:
      return 'danger'
  }
}

const Difficulty = ({level}: { level: DifficultyType }) => (
  <Badge pill color={classForLevel(level)}>
    {level}
  </Badge>
)

export default Difficulty
