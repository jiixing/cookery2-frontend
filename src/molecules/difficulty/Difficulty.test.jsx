import React from 'react'
import renderer from 'react-test-renderer'
import Difficulty from './Difficulty'

describe('components', () => {
  describe('Difficulty', () => {
    it('renders correctly', () => {
      const component = renderer.create(<Difficulty level='easy' />)
      expect(component.toJSON()).toMatchSnapshot()
    })

    it('renders different difficulties', () => {
      const component = renderer.create(<Difficulty level='medium' />)
      expect(component.toJSON()).toMatchSnapshot()
    })
  })
})
