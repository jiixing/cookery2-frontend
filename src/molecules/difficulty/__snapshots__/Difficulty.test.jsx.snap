// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`components Difficulty renders correctly 1`] = `
<span
  className="badge badge-success badge-pill"
>
  easy
</span>
`;

exports[`components Difficulty renders different difficulties 1`] = `
<span
  className="badge badge-warning badge-pill"
>
  medium
</span>
`;
